File: Back-end/src/main/java/com/example/pets/entity/Staff.java
package com.example.pets.entity;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Date;

@Entity
@Table(name = "staff")
@Data
@NoArgsConstructor
public class Staff {
    @Id
    @GeneratedValue
    private Long staffId;
    private String name;
    private String phone;
    private String address;
    private LocalDate birthDate;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "managedShelterId", referencedColumnName = "shelterId")
    private Shelter shelter;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "workAtShelterId", referencedColumnName = "shelterId")
    private Shelter staff;
}

File: Front-end/pet-shelter/src/app/Objects/Staff.ts
export class Staff {
    staffId!:number
    name: string
    phone: string
    email:string
    address: string
    birthDate: Date
    role: string
    shelter: any
    staff!:any
    constructor(name: string, email: string, phone: string, address: string,
        birthDate: Date, role: string, shelterId: any) {
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.address = address;
        this.birthDate = birthDate;
        this.role = role;
        this.shelter = shelterId;
    }
}